core/src/utils/directive.ts

```diff
@@ -223,3 +227,124 @@ export const mergeDirectives =
 			},
 		};
 	};
+
+/**
+ * Properties for configuring server-side rendering directives.
+ */
+export interface SSRDirectiveProps {
+	/**
+	 * Events to be attached to an HTML element.
+	 * @remarks
+	 * Key-value pairs where keys are event types and values are event handlers.
+	 */
+	events?: Partial<Record<keyof HTMLElementEventMap, (event: any) => void>>;
+
+	/**
+	 * Attributes to be added to the provided node.
+	 * @remarks
+	 * The `style` and the `class` attributes must be added separately.
+	 */
+	attributes?: Record<string, AttributeValue | ReadableSignal<AttributeValue>>;
+
+	/**
+	 * Styles to be added to an HTML element.
+	 * @remarks
+	 * Key-value pairs where keys are CSS style properties and values are style values.
+	 */
+	styles?: Partial<Record<keyof CSSStyleDeclaration, StyleValue | ReadableSignal<StyleValue>>>;
+
+	/**
+	 * Class names to be added to an HTML element.
+	 * @remarks
+	 * Key-value pairs where keys are class names and values indicate whether the class should be added (true) or removed (false).
+	 */
+	classNames?: Record<string, boolean | ReadableSignal<boolean>>;
+}
+
+/**
+ * Creates a directive for server-side rendering with bindable elements.
+ * This directive binds events, attributes, styles, and classNames to an HTML element.
+ *
+ * @param propsFn - A function that returns the SSRDirectiveProps with the data to bind.
+ * This function can take an optional parameter that corrspond to the second parameter of the created directive.
+ * @returns A directive object with bound events, attributes, styles, and classNames.
+ */
+export const createSSRDirective =
```

Comment:
```suggestion
export const createAttributesDirective =
```
This is not really linked to SSR.